server/api-util/currency.js

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * Ensures that the given string uses only dots
+ * (e.g. JavaScript floats use dots)
+ *
+ * @param {String} str - string to be formatted
+ *
+ * @return {String} converted string
+ */
+const ensureDotSeparator = str => {
+  return ensureSeparator(str, false);
+};
+
+/**
+ * Convert string to Decimal object (from Decimal.js math library)
+ * Handles both dots and commas as decimal separators
+ *
+ * @param {String} str - string to be converted
+ *
+ * @return {Decimal} numeral value
+ */
+const convertToDecimal = str => {
+  const dotFormattedStr = ensureDotSeparator(str);
+  return new Decimal(dotFormattedStr);
+};
+
+// Divisor can be positive value given as Decimal, Number, or String
+const convertDivisorToDecimal = divisor => {
+  try {
+    const divisorAsDecimal = new Decimal(divisor);
+    if (divisorAsDecimal.isNegative()) {
+      throw new Error(`Parameter (${divisor}) must be a positive number.`);
+    }
+    return divisorAsDecimal;
+  } catch (e) {
+    throw new Error(`Parameter (${divisor}) must present a number.`, e);
+  }
+};
+
+// Detect if the given value is a goog.math.Long object
+// See: https://google.github.io/closure-library/api/goog.math.Long.html
+const isGoogleMathLong = value => {
+  return typeof value === 'object' && isNumber(value.low_) && isNumber(value.high_);
+};
+
+/**
+ * Converts given value to sub unit value and returns it as a number
+ *
+ * @param {Number|String} value
+ *
+ * @param {Decimal|Number|String} subUnitDivisor - should be something that can be converted to
+ * Decimal. (This is a ratio between currency's main unit and sub units.)
+ *
+ * @param {boolean} useComma - optional.
+ * Specify if return value should use comma as separator
+ *
+ * @return {number} converted value
+ */
+exports.convertUnitToSubUnit = (value, subUnitDivisor, useComma = false) => {
+  const subUnitDivisorAsDecimal = convertDivisorToDecimal(subUnitDivisor);
+
+  if (!(typeof value === 'string' || typeof value === 'number')) {
+    throw new TypeError('Value must be either number or string');
+  }
+
+  const val = typeof value === 'string' ? convertToDecimal(value, useComma) : new Decimal(value);
+  const amount = val.times(subUnitDivisorAsDecimal);
+
+  if (!isSafeNumber(amount)) {
+    throw new Error(
+      `Cannot represent money minor unit value ${amount.toString()} safely as a number`
+    );
+  } else if (amount.isInteger()) {
+    return amount.toNumber();
+  } else {
+    throw new Error(`value must divisible by ${subUnitDivisor}`);
+  }
+};
+
+/**
+ * Convert Money to a number
+ *
+ * @param {Money} value
+ *
+ * @return {Number} converted value
+ */
+exports.convertMoneyToNumber = value => {
+  if (!(value instanceof Money)) {
+    throw new Error('Value must be a Money type');
+  }
+  const subUnitDivisorAsDecimal = convertDivisorToDecimal(this.unitDivisor(value.currency));
+  let amount;
+
+  if (isGoogleMathLong(value.amount)) {
+    // TODO: temporarily also handle goog.math.Long values created by
```

Comment:
I'm not sure if this is still valid. It could be that this is just a forgotten piece of code.

Reply:
How can I check if this is still needed or not? I think I have never seen the warning and I tried to check the SDK but I'm not sure how to know if this can be removed or not...